### PR TITLE
ike: set event for multiple server proposals

### PR DESCRIPTION
**Pull request by fhonza**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4433

Describe changes:
- in case of multiple chosen server transforms set an event, log them and don't panic on debug-validate

### CODE REVIEW

**Comment by victorjulien** [comment]
The commit check CI issue looks like something we can ignore.

File: rust/src/ike/ikev1.rs [hunk 1]
```diff
@@ -136,6 +136,11 @@ pub fn handle_ikev1(
                         &tx.hdr.ikev1_header.vendor_ids,
                         &tx.hdr.ikev1_transforms,
                     );
+
+                    if state.ikev1_container.server.transforms.len() > 1 {
```
Comment (catenacyber) [inline rust/src/ike/ikev1.rs:140]: Should we set the event only once we cross the threshold ?
Something like `state.ikev1_container.server.transforms.len() <= 1 ` and `tx.hdr.ikev1_transforms.len() + state.ikev1_container.server.transforms.len()  > 1` before calling update ?

File: rust/src/ike/logger.rs [hunk 2]
```diff
@@ -73,14 +73,19 @@ fn log_ike(
...
-            if index >= 1 {
-                debug_validate_bug_on!(true);
-                break;
+        if state.ikev1_container.server.transforms.len() > 0 {
+            // log the first transform as the chosen one
+            add_attributes(&state.ikev1_container.server.transforms[0], jb)?;
+        }
+        if state.ikev1_container.server.transforms.len() > 1 {
+            // in case we have multiple server transforms log them in a list
+            jb.open_array("server_proposals")?;
```
Comment (catenacyber) [inline rust/src/ike/logger.rs:82]: I wonder if we should have an S-V test for this

**Review by catenacyber — COMMENTED** [review]
I think we would prefer to have only one single event per flow when we cross the threshold, and not one event every time we are over the threshold...
What do you think @satta ?

**Comment by satta** [comment]
> I think we would prefer to have only one single event per flow when we cross the threshold, and not one event every time we are over the threshold...
> What do you think @satta ?

Yes, right. I think one event should be enough when we reach >1 proposals.

File: rust/src/ike/ikev1.rs (hunk 1 above)
Comment (fhonza) [reply rust/src/ike/ikev1.rs:140]: yes, good idea

File: rust/src/ike/logger.rs (hunk 2 above)
Comment (catenacyber) [reply rust/src/ike/logger.rs:82]: Maybe the doc should be updated as well...

Comment (fhonza) [reply rust/src/ike/logger.rs:82]: > I wonder if we should have an S-V test for this

Not sure about that, at least it seems to be a rare case. (not in the rfc and the first pcap I saw with this property)

**Comment by satta** [comment]
> typo in `parameters`

Ah, yep. Will fix that in next PR.

**Pull request closed without merging**